rust/processor/src/db/common/models/emojicoin_models/models/market_24h_rolling_volume.rs

```diff
@@ -0,0 +1,39 @@
+use crate::db::common::models::emojicoin_models::{enums::Period, json_types::EventWithMarket};
+use bigdecimal::BigDecimal;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct RecentOneMinutePeriodicStateEvent {
+    pub market_id: i64,
+    pub market_nonce: i64,
+    pub period_volume: BigDecimal,
+    pub start_time: i64,
+}
+
+// Recent being defined as within the time frame defined in the function logic below.
+impl RecentOneMinutePeriodicStateEvent {
+    pub fn try_from_event(event: EventWithMarket) -> Option<Self> {
+        match event {
+            EventWithMarket::PeriodicState(pse) => {
+                let one_day_ago = chrono::Utc::now() - chrono::Duration::days(1);
+                let one_day_ago_micros = one_day_ago.timestamp_millis() * 1000;
```

Comment:
How about a named constant instead of a magic number?

Reply:
I put the logic for it in `utils` as `one_day_ago_micros()` since it has to be calculated on the fly. My code here was redundant it should've just been `.timestamp_micros()` anyway. I also changed the other place I check for "one day ago" logic and replaced it with the new function

Is that what you meant?